open-hackathon-server/src/hackathon/storage/azure_storage.py

```diff
@@ -22,13 +22,128 @@
+            assert context.get('file_path')
+            file_path = context.file_path
+            result = self.file_service.upload_file_to_azure_from_path(file_path, container_name, blob_name)
+
+        if result is not None:
+            self.cache_manager.invalidate(result)
+        context["url"] = result
+        self.log.debug("File saved at:" + result)
+        return context
+
+    def load(self, context):
+        """Load file from storage
+
+        :type context: Context
+        :param context: the execution context of file loading
+
+        :rtype dict
+        :return the file content
+        """
+        assert context.get("url")
+        print context.url, context.path
+
+        def get_temp():
+            temp_path = self.file_service.download_file_from_azure(azure_file_url=context.url,
+                                                                   local_file_path=context.path)
+            if temp_path is None:
+                return None
+            if context.file_type == FILE_TYPE.TEMPLATE:
+                with open(temp_path) as template_file:
+                    return json.load(template_file)
+            else:
+                return None
+
+        return self.cache_manager.get_cache(key=context.url, createfunc=get_temp)
+
+    def delete(self, context):
+        """Delete file from Azure storage
+
+        :type context: Context
+        :param context: must have {"url":""} key or {"container_name":"***","blob_name":"***"} keys
+
+        :rtype bool
+        :return True if successfully deleted else False
+        """
+        try:
+            if context.get("url"):
+                url_arr = context.url.split('/')
+                file_name = url_arr[-1]
+                hackathon_name = url_arr[-2]
+                container_name = url_arr[-3]
+                blob_name = hackathon_name + "/" + file_name
+                print container_name, blob_name
+            else:
+                assert context.get("container_name") and context.get("blob_name")
+                container_name = context.container_name
+                blob_name = context.blob_name
+            self.file_service.delete_file_from_azure(container_name, blob_name)
+            url = self.file_service.blob_service.make_blob_url(container_name, blob_name)
+            self.cache_manager.invalidate(url)
+            return True
+        except Exception as e:
+            self.log.error(e)
+            return None
+
+    def report_health(self):
+        """The status of Azure storage should be always True"""
+        return {
+            HEALTH.STATUS: HEALTH_STATUS.OK
+        }
 
-    def __init__(self):
-        return
\ No newline at end of file
+    def __generate_file_name(self, file_name):
+        """refresh file_name = hack_name + uuid(10) + time + suffix
+        """
+        suffix = file_name.split('.')[-1]
+        real_name = g.hackathon.name + "/" + str(uuid1())[0:9] + strftime("%Y%m%d%H%M%S") + "." + suffix
```

Comment:
It's not recommended that `g` is used here. `g` depends on a http request but the storage should have no knowledge about the request.